api/comment_threads.rb

```diff
@@ -27,9 +27,12 @@
     error 404, [t(:requested_object_not_found)].to_json
   end
 
-  if params["user_id"] and bool_mark_as_read
+  if params["user_id"]
     user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
-    user.mark_as_read(thread) if user
+  end
+
+  if user and bool_mark_as_read
+    user.mark_as_read(thread)
   end
```

Comment:
What does this change accomplish?  

It looks like the overall behavior is the same, but the `bool_mark_as_read` check is moved to the second loop, which means that sometimes we are fetching the user from the database even if we don't need to.  

It also looks like the `if user` check can happen even if we never assigned anything to `user` in the first place.  In python, this would cause a `NameError`.  Is it okay to do that in Ruby?  (In other words, what happens if you don't pass a "user_id"? 

Also, do we just accept whatever user_id parameter gets passed in the URL?  Seems like there might be a minor security risk there.  (Can a student mark a thread as already having been read by another student?)